### PR TITLE
Basic mobs can be sentience & mind transfer potioned.  + simple access component

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request
This PR makes it possible to sentience potion & mind transfer potion basic mobs.

Because the syndicate sentientience potion also added an abstract ID to the mob, and basic mobs did not support this, i've created a new component that holds this access.

The abductor all access gland has been refactored to use this component.

## Why It's Good For The Game

If basic mobs are going to replace simple_animals, they need to support one of the biggest features that interacts with simple_animals, sentience granters such as the sentience potion.

Currently, there are some basic mobs that would be fun sentience targets, moonicorns and hauberoaches come to mind and going forward, as i add more cytology mobs, i'd rather make them basic mobs than make them simple_animals which should be considered a deprecated type.
 
The new component also allows coders to grant access not tied to an ID card, but that can optionally be tied to an organ or implant.

## Changelog

:cl:
fix: Cows, roaches and their subtypes can be sentienced and mind transfer potioned again.
refactor: refactored syndicate sentience potion and abductor all access gland to use the new simple_access component.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
how does simple access interact with ai controller pathfinding?

**Comment by Krysonism** [comment]
> how does simple access interact with ai controller pathfinding?

No clue. It is limited to syndicate sentience potion and all access gland so far.

**Comment by Fikou** [comment]
> > how does simple access interact with ai controller pathfinding?
> 
> No clue. It is limited to syndicate sentience potion and all access gland so far.

i think it checks for an abstract id card you have to pass onto the pathfinder, would be nice if you could make it compatible with the compotent

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -686,58 +686,62 @@
...
+			to_chat(user, span_warning("[src] won't work on [basic_dumb_bitch]."))
+			return
+
+	to_chat(user, span_notice("You offer [src] to [dumb_mob]..."))
 	being_used = TRUE
 
-	var/list/candidates = poll_candidates_for_mob("Do you want to play as [SM.name]?", ROLE_SENTIENCE, ROLE_SENTIENCE, 5 SECONDS, SM, POLL_IGNORE_SENTIENCE_POTION) // see poll_ignore.dm
+	var/list/candidates = poll_candidates_for_mob("Do you want to play as [dumb_mob.name]?", ROLE_SENTIENCE, ROLE_SENTIENCE, 5 SECONDS, dumb_mob, POLL_IGNORE_SENTIENCE_POTION) // see poll_ignore.dm
 	if(LAZYLEN(candidates))
 		var/mob/dead/observer/C = pick(candidates)
```
Comment (SuperNovaa41) [inline code/modules/research/xenobiology/xenobiology.dm:714]: Not required but, you're already changing every other var name you're touching, why not this one?

File: code/datums/components/simple_access.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/datum/component/simple_access/Initialize(list/new_access, atom/donor_atom)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	access = new_access
+	RegisterSignal(parent, COMSIG_MOB_TRIED_ACCESS, .proc/on_tried_access)
+	if(!donor_atom)
+		return
+	if(istype(donor_atom, /obj/item/organ))
+		RegisterSignal(donor_atom, COMSIG_ORGAN_REMOVED, .proc/on_donor_removed)
```
Comment (Kylerace) [inline code/datums/components/simple_access.dm:16]: should also register to donor being deleted

File: code/modules/jobs/access.dm [hunk 3]
```diff
@@ -13,21 +13,21 @@
 	if(isAdminGhostAI(accessor))
 		//Access can't stop the abuse
 		return TRUE
-	else if(istype(accessor) && SEND_SIGNAL(accessor, COMSIG_MOB_ALLOWED, src))
+	//If the mob has the simple_access component with the requried access, we let them in.
+	else if(SEND_SIGNAL(accessor, COMSIG_MOB_TRIED_ACCESS, src) & ACCESS_ALLOWED)
```
Comment (Kylerace) [inline code/modules/jobs/access.dm:17]: why are you creating a new signal when COMSIG_OBJ_ALLOWED already exists and seems to have the same purpose?

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/jobs/access.dm (hunk 3 above)
Comment (Krysonism) [reply code/modules/jobs/access.dm:17]: I've renamed COMSIG_MOB_ALLOWED.

File: code/datums/components/simple_access.dm (hunk 2 above)
Comment (Krysonism) [reply code/datums/components/simple_access.dm:16]: Good catch

**Pull request merged**